charts/datadog/templates/_helpers.tpl

```diff
@@ -365,9 +365,26 @@ false
 {{- end -}}
 
 {{/*
-Return true if a APM over UDS is configured.
+Return true hostPath should be use for DSD socket. Return always false on GKE autopilot.
+*/}}
+{{- define "should-mount-hostPath-for-dsd-socket" -}}
+{{- if or .Values.providers.gke.autopilot (eq .Values.targetSystem "windows") -}}
+false
+{{- end -}}
+{{- if .Values.datadog.dogstatsd.useSocketVolume -}}
+false
+{{- else -}}
+true
+{{- end -}}
```

Comment:
According to the values.yaml file, useSocketVolume (by default true) indicates that a HostVolume should be created for the dogstatsd socket directory. After this PR, by default the dsdsocket is an EmptyDir. Isn't that a bug ?

Reply:
Hello,
Thanks for catching this Bug. the PR #363 should fix it.